ci/ray_ci/doc/update_cache_env.py

```diff
@@ -0,0 +1,89 @@
+import pickle
+from sphinx.project import Project
+import os
+import time
+from typing import List
+from datetime import datetime, timezone
+import click
+
+
+def list_pending_files(ray_dir: str) -> List[str]:
+    """List all files that are added/modified in git repo."""
+    pending_files = []
+    with open(f"{ray_dir}/pending_files.txt", "r") as f:
+        pending_files = f.readlines()
+        pending_files = [file.strip() for file in pending_files]
+    return pending_files
+
+
+def update_environment_pickle(ray_dir: str, pending_files: List[str]) -> None:
+    """
+    Update the environment pickle file with
+    new source and doctree directory, and modify source file timestamps.
+    """
+    ray_doc_dir = os.path.join(ray_dir, "doc")
+    with open(
+        os.path.join(ray_doc_dir, "_build/doctrees/environment.pickle"), "rb"
```

Comment:
the `_build/doctrees/environment.pickle` is defined as a constant elsewhere (in build_cache i think), maybe just import it here

Reply:
done